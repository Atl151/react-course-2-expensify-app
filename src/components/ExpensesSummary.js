import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';

const getMessage = (expenses) => {
    const length = expenses.length;
    const total = numeral(getExpensesTotal(expenses) / 100).format('$0,0.00');

    if(length === 1){
        return `Viewing ${length} expense totalling ${total}`;
    } else {
        return `Viewing ${length} expenses totalling ${total}`
    }
}

export const ExpensesSummary = (props) => (
    <div>
        <h3>Summary</h3>
        <p>{getMessage(props.expenses)}</p>
    </div>
);

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);