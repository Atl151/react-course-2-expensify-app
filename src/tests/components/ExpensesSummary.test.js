import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render correct message for one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[1]]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render correct message for multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});
