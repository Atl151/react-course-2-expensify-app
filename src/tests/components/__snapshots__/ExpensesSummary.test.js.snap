// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct message for multiple expenses 1`] = `
<div>
  <h3>
    Summary
  </h3>
  <p>
    Viewing 3 expenses totalling $1,141.95
  </p>
</div>
`;

exports[`should render correct message for one expense 1`] = `
<div>
  <h3>
    Summary
  </h3>
  <p>
    Viewing 1 expense totalling $1,095.00
  </p>
</div>
`;
